File: android/app/src/main/java/com/nearme/app/ui/notifications/NotificationComponents.kt
package com.nearme.app.ui.notifications

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.nearme.app.models.NotificationItem
import com.nearme.app.models.NotificationAction
import com.nearme.app.models.NotificationFilter
import com.nearme.app.models.SnoozeDuration
import java.text.SimpleDateFormat
import java.util.*

// MARK: - Notification Card Component
@Composable
fun NotificationCard(
    notification: NotificationItem,
    onAction: (NotificationAction) -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 4.dp),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
        shape = RoundedCornerShape(12.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            // Header with title and timestamp
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.Top
            ) {
                Column(
                    modifier = Modifier.weight(1f)
                ) {
                    Text(
                        text = notification.title,
                        style = MaterialTheme.typography.bodyLarge,
                        fontWeight = FontWeight.Medium,
                        color = MaterialTheme.colorScheme.onSurface
                    )
                    
                    Spacer(modifier = Modifier.height(4.dp))
                    
                    Text(
                        text = notification.body,
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        maxLines = 2,
                        overflow = TextOverflow.Ellipsis
                    )
                }
                
                Column(
                    horizontalAlignment = Alignment.End
                ) {
                    Text(
                        text = formatTimestamp(notification.timestamp),
                        style = MaterialTheme.typography.labelSmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    
                    Spacer(modifier = Modifier.height(4.dp))
                    
                    if (notification.isRead) {
                        Icon(
                            imageVector = Icons.Default.CheckCircle,
                            contentDescription = "Read",
                            tint = MaterialTheme.colorScheme.primary,
                            modifier = Modifier.size(16.dp)
                        )
                    }
                }
            }
            
            // Action buttons
            if (notification.actions.isNotEmpty()) {
                Spacer(modifier = Modifier.height(12.dp))
                
                Row(
                    horizontalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    notification.actions.forEach { action ->
                        NotificationActionButton(
                            action = action,
                            onTap = { onAction(action) }
                        )
                    }
                }
            }
        }
    }
}

// MARK: - Notification Action Button
@Composable
fun NotificationActionButton(
    action: NotificationAction,
    onTap: () -> Unit,
    modifier: Modifier = Modifier
) {
    Button(
        onClick = onTap,
        modifier = modifier,
        colors = ButtonDefaults.buttonColors(
            containerColor = action.backgroundColor,
            contentColor = action.foregroundColor
        ),
        shape = RoundedCornerShape(8.dp),
        contentPadding = PaddingValues(horizontal = 12.dp, vertical = 6.dp)
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(4.dp)
        ) {
            Icon(
                imageVector = action.icon,
                contentDescription = null,
                modifier = Modifier.size(14.dp)
            )
            
            Text(
                text = action.title,
                style = MaterialTheme.typography.labelSmall,
                fontWeight = FontWeight.Medium
            )
        }
    }
}

// MARK: - Notification History Screen
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun NotificationHistoryScreen(
    viewModel: NotificationHistoryViewModel,
    modifier: Modifier = Modifier
) {
    val notifications by viewModel.filteredNotifications.collectAsState()
    val selectedFilter by viewModel.selectedFilter.collectAsState()
    val isLoading by viewModel.isLoading.collectAsState()
    
    Column(modifier = modifier.fillMaxSize()) {
        // Top App Bar
        TopAppBar(
            title = { Text("Notifications") },
            actions = {
                TextButton(
                    onClick = { viewModel.markAllAsRead() }
                ) {
                    Text("Mark All Read")
                }
            }
        )
        
        // Filter tabs
        ScrollableTabRow(
            selectedTabIndex = NotificationFilter.values().indexOf(selectedFilter),
            modifier = Modifier.fillMaxWidth()
        ) {
            NotificationFilter.values().forEachIndexed { index, filter ->
                Tab(
                    selected = selectedFilter == filter,
                    onClick = { viewModel.selectedFilter = filter },
                    text = { Text(filter.title) }
                )
            }
        }
        
        Divider()
        
        // Notification list
        if (isLoading) {
            Box(
                modifier = Modifier.fillMaxSize(),
                contentAlignment = Alignment.Center
            ) {
                CircularProgressIndicator()
            }
        } else if (notifications.isEmpty()) {
            EmptyNotificationState(filter = selectedFilter)
        } else {
            LazyColumn(
                modifier = Modifier.fillMaxSize(),
                contentPadding = PaddingValues(vertical = 8.dp)
            ) {
                items(notifications) { notification ->
                    NotificationCard(
                        notification = notification,
                        onAction = { action ->
                            viewModel.handleAction(action, notification)
                        }
                    )
                }
            }
        }
    }
}

// MARK: - Empty Notification State
@Composable
fun EmptyNotificationState(
    filter: NotificationFilter,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(32.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Icon(
            imageVector = filter.emptyStateIcon,
            contentDescription = null,
            modifier = Modifier.size(48.dp),
            tint = MaterialTheme.colorScheme.onSurfaceVariant
        )
        
        Spacer(modifier = Modifier.height(16.dp))
        
        Text(
            text = filter.emptyStateTitle,
            style = MaterialTheme.typography.headlineSmall,
            fontWeight = FontWeight.Medium,
            color = MaterialTheme.colorScheme.onSurface
        )
        
        Spacer(modifier = Modifier.height(8.dp))
        
        Text(
            text = filter.emptyStateMessage,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = androidx.compose.ui.text.style.TextAlign.Center
        )
    }
}

// MARK: - Notification Settings Screen
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun NotificationSettingsScreen(
    viewModel: NotificationSettingsViewModel,
    notificationManager: com.nearme.app.notifications.NotificationManager,
    modifier: Modifier = Modifier
) {
    val settings by viewModel.settings.collectAsState()
    val isNotificationEnabled by notificationManager.isNotificationEnabled.collectAsState()
    
    Column(modifier = modifier.fillMaxSize()) {
        TopAppBar(
            title = { Text("Notification Settings") }
        )
        
        LazyColumn(
            modifier = Modifier.fillMaxSize(),
            contentPadding = PaddingValues(16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            // Permission Status Section
            item {
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    colors = CardDefaults.cardColors(
                        containerColor = MaterialTheme.colorScheme.surfaceVariant
                    )
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(16.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(
                                text = "Notifications",
                                style = MaterialTheme.typography.bodyLarge,
                                fontWeight = FontWeight.Medium
                            )
                            
                            Text(
                                text = if (isNotificationEnabled) "Enabled" else "Disabled - Enable in Settings",
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                        
                        if (!isNotificationEnabled) {
                            Button(
                                onClick = { notificationManager.requestNotificationPermission() }
                            ) {
                                Text("Enable")
                            }
                        }
                    }
                }
            }
            
            // Notification Preferences Section
            item {
                Text(
                    text = "Notification Types",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Medium,
                    modifier = Modifier.padding(vertical = 8.dp)
                )
            }
            
            item {
                SettingsSwitchItem(
                    title = "Location Reminders",
                    subtitle = "Get notified when you're near places",
                    checked = settings.locationRemindersEnabled,
                    onCheckedChange = { viewModel.updateLocationRemindersEnabled(it) }
                )
            }
            
            item {
                SettingsSwitchItem(
                    title = "Task Completion Alerts",
                    subtitle = "Get notified when tasks are completed",
                    checked = settings.taskCompletionAlerts,
                    onCheckedChange = { viewModel.updateTaskCompletionAlerts(it) }
                )
            }
            
            item {
                SettingsSwitchItem(
                    title = "Daily Summary",
                    subtitle = "Receive a daily summary of your tasks",
                    checked = settings.dailySummary,
                    onCheckedChange = { viewModel.updateDailySummary(it) }
                )
            }
            
            // Quiet Hours Section
            item {
                Text(
                    text = "Quiet Hours",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Medium,
                    modifier = Modifier.padding(vertical = 8.dp)
                )
            }
            
            item {
                SettingsSwitchItem(
                    title = "Enable Quiet Hours",
                    subtitle = "Notifications will be silenced during these hours",
                    checked = settings.quietHoursEnabled,
                    onCheckedChange = { viewModel.updateQuietHoursEnabled(it) }
                )
            }
            
            if (settings.quietHoursEnabled) {
                item {
                    SettingsTimePickerItem(
                        title = "Start Time",
                        time = settings.quietHoursStart,
                        onTimeChange = { viewModel.updateQuietHoursStart(it) }
                    )
                }
                
                item {
                    SettingsTimePickerItem(
                        title = "End Time",
                        time = settings.quietHoursEnd,
                        onTimeChange = { viewModel.updateQuietHoursEnd(it) }
                    )
                }
            }
            
            // Snooze Defaults Section
            item {
                Text(
                    text = "Snooze Settings",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Medium,
                    modifier = Modifier.padding(vertical = 8.dp)
                )
            }
            
            item {
                SettingsDropdownItem(
                    title = "Default Snooze Duration",
                    subtitle = "Default duration when snoozing notifications",
                    selectedValue = settings.defaultSnoozeDuration.title,
                    onValueChange = { viewModel.updateDefaultSnoozeDuration(it) }
                )
            }
        }
    }
}

// MARK: - Settings Components
@Composable
fun SettingsSwitchItem(
    title: String,
    subtitle: String,
    checked: Boolean,
    onCheckedChange: (Boolean) -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surface
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyLarge,
                    fontWeight = FontWeight.Medium
                )
                
                Text(
                    text = subtitle,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
            
            Switch(
                checked = checked,
                onCheckedChange = onCheckedChange
            )
        }
    }
}

@Composable
fun SettingsTimePickerItem(
    title: String,
    time: Date,
    onTimeChange: (Date) -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surface
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.bodyLarge,
                fontWeight = FontWeight.Medium
            )
            
            Text(
                text = SimpleDateFormat("HH:mm", Locale.getDefault()).format(time),
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.primary
            )
        }
    }
}

@Composable
fun SettingsDropdownItem(
    title: String,
    subtitle: String,
    selectedValue: String,
    onValueChange: (SnoozeDuration) -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surface
        )
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.bodyLarge,
                fontWeight = FontWeight.Medium
            )
            
            Text(
                text = subtitle,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
            
            Spacer(modifier = Modifier.height(8.dp))
            
            Text(
                text = selectedValue,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.primary
            )
        }
    }
}

// MARK: - Helper Functions
private fun formatTimestamp(timestamp: Date): String {
    val now = Date()
    val diff = now.time - timestamp.time
    
    return when {
        diff < 60 * 1000 -> "Just now"
        diff < 60 * 60 * 1000 -> "${diff / (60 * 1000)}m ago"
        diff < 24 * 60 * 60 * 1000 -> "${diff / (60 * 60 * 1000)}h ago"
        else -> SimpleDateFormat("MMM d", Locale.getDefault()).format(timestamp)
    }
}

File: NearMe/Components/NotificationComponents.swift
import SwiftUI
import UserNotifications

// MARK: - Notification Card Component
struct NotificationCard: View {
    let notification: NotificationItem
    let onAction: (NotificationAction) -> Void
    
    var body: some View {
        VStack(alignment: .leading, spacing: DesignSystem.Spacing.sm) {
            // Header with title and timestamp
            HStack {
                VStack(alignment: .leading, spacing: DesignSystem.Spacing.xs) {
                    Text(notification.title)
                        .font(DesignSystem.Typography.bodyEmphasized)
                        .foregroundColor(DesignSystem.Colors.textPrimary)
                    
                    Text(notification.body)
                        .font(DesignSystem.Typography.callout)
                        .foregroundColor(DesignSystem.Colors.textSecondary)
                        .lineLimit(2)
                }
                
                Spacer()
                
                VStack(alignment: .trailing, spacing: DesignSystem.Spacing.xs) {
                    Text(notification.timestamp.formatted(date: .omitted, time: .shortened))
                        .font(DesignSystem.Typography.caption1)
                        .foregroundColor(DesignSystem.Colors.textTertiary)
                    
                    if notification.isRead {
                        Circle()
                            .fill(DesignSystem.Colors.success)
                            .frame(width: 8, height: 8)
                    }
                }
            }
            
            // Action buttons
            if !notification.actions.isEmpty {
                HStack(spacing: DesignSystem.Spacing.sm) {
                    ForEach(notification.actions, id: \.identifier) { action in
                        NotificationActionButton(
                            action: action,
                            onTap: { onAction(action) }
                        )
                    }
                }
            }
        }
        .padding(DesignSystem.Spacing.md)
        .background(
            RoundedRectangle(cornerRadius: DesignSystem.CornerRadius.md)
                .fill(DesignSystem.Colors.card)
        )
        .designSystemShadow(DesignSystem.Shadow.small)
    }
}

// MARK: - Notification Action Button
struct NotificationActionButton: View {
    let action: NotificationAction
    let onTap: () -> Void
    
    var body: some View {
        Button(action: onTap) {
            HStack(spacing: DesignSystem.Spacing.xs) {
                Image(systemName: action.icon)
                    .font(.system(size: 14, weight: .medium))
                
                Text(action.title)
                    .font(DesignSystem.Typography.caption1)
                    .fontWeight(.medium)
            }
            .padding(.horizontal, DesignSystem.Spacing.sm)
            .padding(.vertical, DesignSystem.Spacing.xs)
            .background(
                RoundedRectangle(cornerRadius: DesignSystem.CornerRadius.sm)
                    .fill(action.backgroundColor)
            )
            .foregroundColor(action.foregroundColor)
        }
    }
}

// MARK: - Notification History View
struct NotificationHistoryView: View {
    @StateObject private var viewModel = NotificationHistoryViewModel()
    @EnvironmentObject var notificationManager: NotificationManager
    
    var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Filter tabs
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: DesignSystem.Spacing.sm) {
                        ForEach(NotificationFilter.allCases, id: \.self) { filter in
                            FilterTab(
                                filter: filter,
                                isSelected: viewModel.selectedFilter == filter,
                                onTap: { viewModel.selectedFilter = filter }
                            )
                        }
                    }
                    .padding(.horizontal, DesignSystem.Spacing.md)
                }
                .padding(.vertical, DesignSystem.Spacing.sm)
                
                Divider()
                
                // Notification list
                if viewModel.filteredNotifications.isEmpty {
                    EmptyNotificationState(filter: viewModel.selectedFilter)
                } else {
                    ScrollView {
                        LazyVStack(spacing: DesignSystem.Spacing.sm) {
                            ForEach(viewModel.filteredNotifications) { notification in
                                NotificationCard(
                                    notification: notification,
                                    onAction: { action in
                                        viewModel.handleAction(action, for: notification)
                                    }
                                )
                            }
                        }
                        .padding(DesignSystem.Spacing.md)
                    }
                }
            }
            .navigationTitle("Notifications")
            .navigationBarTitleDisplayMode(.large)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button("Mark All Read") {
                        viewModel.markAllAsRead()
                    }
                    .font(DesignSystem.Typography.callout)
                    .foregroundColor(DesignSystem.Colors.primary)
                }
            }
            .onAppear {
                viewModel.loadNotifications()
            }
        }
    }
}

// MARK: - Filter Tab
struct FilterTab: View {
    let filter: NotificationFilter
    let isSelected: Bool
    let onTap: () -> Void
    
    var body: some View {
        Button(action: onTap) {
            Text(filter.title)
                .font(DesignSystem.Typography.callout)
                .fontWeight(isSelected ? .semibold : .medium)
                .foregroundColor(isSelected ? DesignSystem.Colors.textInverse : DesignSystem.Colors.textSecondary)
                .padding(.horizontal, DesignSystem.Spacing.md)
                .padding(.vertical, DesignSystem.Spacing.sm)
                .background(
                    RoundedRectangle(cornerRadius: DesignSystem.CornerRadius.lg)
                        .fill(isSelected ? DesignSystem.Colors.primary : DesignSystem.Colors.surface)
                )
        }
    }
}

// MARK: - Empty Notification State
struct EmptyNotificationState: View {
    let filter: NotificationFilter
    
    var body: some View {
        VStack(spacing: DesignSystem.Spacing.lg) {
            Image(systemName: filter.emptyStateIcon)
                .font(.system(size: 48, weight: .light))
                .foregroundColor(DesignSystem.Colors.textTertiary)
            
            VStack(spacing: DesignSystem.Spacing.sm) {
                Text(filter.emptyStateTitle)
                    .font(DesignSystem.Typography.title3)
                    .foregroundColor(DesignSystem.Colors.textPrimary)
                
                Text(filter.emptyStateMessage)
                    .font(DesignSystem.Typography.body)
                    .foregroundColor(DesignSystem.Colors.textSecondary)
                    .multilineTextAlignment(.center)
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .padding(DesignSystem.Spacing.xl)
    }
}

// MARK: - Notification Settings View
struct NotificationSettingsView: View {
    @EnvironmentObject var notificationManager: NotificationManager
    @StateObject private var viewModel = NotificationSettingsViewModel()
    
    var body: some View {
        NavigationView {
            Form {
                // Permission Status Section
                Section {
                    HStack {
                        VStack(alignment: .leading, spacing: DesignSystem.Spacing.xs) {
                            Text("Notifications")
                                .font(DesignSystem.Typography.bodyEmphasized)
                            
                            Text(permissionStatusText)
                                .font(DesignSystem.Typography.callout)
                                .foregroundColor(DesignSystem.Colors.textSecondary)
                        }
                        
                        Spacer()
                        
                        if !notificationManager.isNotificationEnabled {
                            Button("Enable") {
                                notificationManager.requestNotificationPermission()
                            }
                            .font(DesignSystem.Typography.callout)
                            .foregroundColor(DesignSystem.Colors.primary)
                        }
                    }
                    .padding(.vertical, DesignSystem.Spacing.xs)
                } header: {
                    Text("Permission Status")
                }
                
                // Notification Preferences Section
                Section {
                    Toggle("Location Reminders", isOn: $viewModel.locationRemindersEnabled)
                    Toggle("Task Completion Alerts", isOn: $viewModel.taskCompletionAlerts)
                    Toggle("Daily Summary", isOn: $viewModel.dailySummary)
                } header: {
                    Text("Notification Types")
                } footer: {
                    Text("Choose which types of notifications you'd like to receive")
                }
                
                // Quiet Hours Section
                Section {
                    Toggle("Enable Quiet Hours", isOn: $viewModel.quietHoursEnabled)
                    
                    if viewModel.quietHoursEnabled {
                        DatePicker("Start Time", selection: $viewModel.quietHoursStart, displayedComponents: .hourAndMinute)
                        DatePicker("End Time", selection: $viewModel.quietHoursEnd, displayedComponents: .hourAndMinute)
                    }
                } header: {
                    Text("Quiet Hours")
                } footer: {
                    Text("Notifications will be silenced during these hours")
                }
                
                // Snooze Defaults Section
                Section {
                    Picker("Default Snooze Duration", selection: $viewModel.defaultSnoozeDuration) {
                        ForEach(SnoozeDuration.allCases, id: \.self) { duration in
                            Text(duration.title).tag(duration)
                        }
                    }
                } header: {
                    Text("Snooze Settings")
                } footer: {
                    Text("Default duration when snoozing notifications")
                }
            }
            .navigationTitle("Notification Settings")
            .navigationBarTitleDisplayMode(.large)
            .onAppear {
                viewModel.loadSettings()
            }
            .onDisappear {
                viewModel.saveSettings()
            }
        }
    }
    
    private var permissionStatusText: String {
        switch notificationManager.authorizationStatus {
        case .authorized:
            return "Enabled"
        case .denied:
            return "Disabled - Enable in Settings"
        case .notDetermined:
            return "Not Set"
        case .provisional:
            return "Provisional"
        case .ephemeral:
            return "Ephemeral"
        @unknown default:
            return "Unknown"
        }
    }
}

// MARK: - Notification Permission Education View
struct NotificationPermissionEducationView: View {
    let onRequestPermission: () -> Void
    let onSkip: () -> Void
    
    var body: some View {
        VStack(spacing: DesignSystem.Spacing.xl) {
            Spacer()
            
            // Icon
            Image(systemName: "bell.badge")
                .font(.system(size: 64, weight: .light))
                .foregroundColor(DesignSystem.Colors.primary)
            
            // Content
            VStack(spacing: DesignSystem.Spacing.md) {
                Text("Stay on Track")
                    .font(DesignSystem.Typography.largeTitle)
                    .fontWeight(.bold)
                    .foregroundColor(DesignSystem.Colors.textPrimary)
                
                Text("Get reminded when you're near the right places. We'll send you helpful notifications to keep you organized.")
                    .font(DesignSystem.Typography.body)
                    .foregroundColor(DesignSystem.Colors.textSecondary)
                    .multilineTextAlignment(.center)
                    .lineLimit(nil)
            }
            
            // Benefits
            VStack(spacing: DesignSystem.Spacing.md) {
                BenefitRow(
                    icon: "location.circle",
                    title: "Location-Based Reminders",
                    description: "Get notified when you arrive at specific places"
                )
                
                BenefitRow(
                    icon: "clock",
                    title: "Smart Timing",
                    description: "Notifications appear at the right moment"
                )
                
                BenefitRow(
                    icon: "checkmark.circle",
                    title: "Quick Actions",
                    description: "Complete, snooze, or mute tasks directly from notifications"
                )
            }
            
            Spacer()
            
            // Action buttons
            VStack(spacing: DesignSystem.Spacing.md) {
                PrimaryButton(
                    title: "Enable Notifications",
                    action: onRequestPermission
                )
                
                SecondaryButton(
                    title: "Skip for Now",
                    action: onSkip
                )
            }
        }
        .padding(DesignSystem.Spacing.xl)
        .background(DesignSystem.Colors.background)
    }
}

// MARK: - Benefit Row
struct BenefitRow: View {
    let icon: String
    let title: String
    let description: String
    
    var body: some View {
        HStack(spacing: DesignSystem.Spacing.md) {
            Image(systemName: icon)
                .font(.system(size: 24, weight: .medium))
                .foregroundColor(DesignSystem.Colors.primary)
                .frame(width: 32, height: 32)
            
            VStack(alignment: .leading, spacing: DesignSystem.Spacing.xs) {
                Text(title)
                    .font(DesignSystem.Typography.bodyEmphasized)
                    .foregroundColor(DesignSystem.Colors.textPrimary)
                
                Text(description)
                    .font(DesignSystem.Typography.callout)
                    .foregroundColor(DesignSystem.Colors.textSecondary)
            }
            
            Spacer()
        }
    }
}

// MARK: - Previews
struct NotificationComponents_Previews: PreviewProvider {
    static var previews: some View {
        Group {
            // Notification Card Preview
            NotificationCard(
                notification: NotificationItem.sample,
                onAction: { _ in }
            )
            .padding()
            .background(DesignSystem.Colors.background)
            
            // Notification History Preview
            NotificationHistoryView()
                .environmentObject(NotificationManager())
            
            // Notification Settings Preview
            NotificationSettingsView()
                .environmentObject(NotificationManager())
            
            // Permission Education Preview
            NotificationPermissionEducationView(
                onRequestPermission: {},
                onSkip: {}
            )
        }
    }
}
